src/types/check_config.rs

```diff
@@ -91,17 +91,35 @@ impl CheckConfig {
             .map(|c| first_slot + (interval_secs * c))
             .collect()
     }
+
+    pub fn should_run(&self, tick: Tick, current_region: String) -> bool {
+        // Determines if this check should run in the current region at this current tick
+
+        // If region configs aren't present, assume we should always run this check
+        if self.region_schedule_mode.is_none() {
+            return true;
+        }
+
+        let Some(active_regions) = self.active_regions.as_ref() else {
+            return true;
+        };
+
+        let running_region_idx = (tick.time().timestamp() / self.interval as i64)
+            .checked_rem(active_regions.len() as i64)
+            .unwrap();
+        active_regions[running_region_idx as usize] == current_region
+    }
 }
 
 #[cfg(test)]
 mod tests {
-    use chrono::TimeDelta;
-    use similar_asserts::assert_eq;
-    use uuid::{uuid, Uuid};
-
     use super::{CheckConfig, CheckInterval, RequestMethod};
+    use crate::config_store::Tick;
     use crate::types::check_config::MAX_CHECK_INTERVAL_SECS;
     use crate::types::shared::RegionScheduleMode;
+    use chrono::{DateTime, TimeDelta, Utc};
+    use similar_asserts::assert_eq;
+    use uuid::{uuid, Uuid};
```

Comment:
Ah your IDE is re-ordering these I think. All good, kinda unfortunate rn there's no way to keep these consistent